### PR TITLE
Add new package. react-input-mask-format

### PR DESCRIPTION
**Pull request by Temirtator**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test react-input-mask-format`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by Temirtator** [comment]
@anuti @FabienLavocat @KentarouTakeda @larrybahr @mernen @josefpaij @danmana @guillaume-ro-fr @archy-bold @braincore @frabnt @alexdor @mahnuh @Conrad777 @adripanico @wertzui @lekoaf @ElianCordoba @indigolain @ricardo-mello @rnicholus @mrjack88 @canoceto @nobu222 @Marcoru97 @tonybadguy @Ilmarinen100 @IVIosi @denisname @chenzhutian @russellporter

please can some of you review my PR?

File: types/react-input-mask-format/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
+// Type definitions for react-input-mask-format 1.0
+// Project: https://github.com/temirtator/react-input-mask-format
+// Definitions by: Alexandre Paré <https://github.com/apare>
```
Comment (amcasey) [inline types/react-input-mask-format/index.d.ts:3]: Do these people know they're being signed up as maintainers?  Should you be in the list?

File: types/react-input-mask-format/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+   * * currentState: Current raw input state. Not defined during component render.
+   * * nextState: Input state with applied mask. Contains value and selection fields.
+   */
+  beforeMaskedStateChange?(states: BeforeMaskedStateChangeStates): InputState;
+}
+
+export class ReactInputMask extends React.Component<Props> {
+}
+
+export default ReactInputMask;
```
Comment (amcasey) [inline types/react-input-mask-format/index.d.ts:65]: It doesn't appear to be a default export in the underlying library?

File: types/react-input-mask-format/tslint.json [hunk 3]
```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "npm-naming": [true,{"mode":"code","errors":[["NeedsExportEquals",false]]}]
```
Comment (amcasey) [inline types/react-input-mask-format/tslint.json:4]: Please use line-level suppressions, rather than global suppressions.

**Review by amcasey — CHANGES_REQUESTED** [review]

**Comment by Temirtator** [comment]
@amcasey fixed

**Comment by lekoaf** [comment]
You're the maintainer and creator of the underlying package. Why not just write it in typescript instead of adding a type definition here?

**Comment by Temirtator** [comment]
@lekoaf hi! I cloned this project from react-input-type. I decided to maintain this repository by myself, because developer of react-input-mask didn't respond for my PR. I just need fast solution. Maybe later, I will move all code to typescript. Thanks

**Comment by lekoaf** [comment]
> @lekoaf hi! I cloned this project from react-input-type. I decided to maintain this repository by myself, because developer of react-input-mask didn't respond for my PR. I just need fast solution. Maybe later, I will move all code to typescript. Thanks

Just add this as an `index.d.ts` to your code or something and make sure it's published.

And why was I even pinged in this PR?

**Comment by Temirtator** [comment]
@lekoaf  are you DT maintainer?

File: types/react-input-mask-format/index.d.ts [hunk 4]
```diff
@@ -1,7 +1,6 @@
 // Type definitions for react-input-mask-format 1.0
 // Project: https://github.com/temirtator/react-input-mask-format
 // Definitions by: Temirlan Shagyrov <https://github.com/temirtator>
-//                 Nikita Lobachev <https://github.com/sanniassin>
```
Comment (Temirtator) [inline types/react-input-mask-format/index.d.ts]: Since I decided to maintain the source code myself, I think it would be correct to write only myself.

**Comment by lekoaf** [comment]
> @lekoaf are you DT maintainer?

No, but I was pinged before you edited your comment.

Screenshot from 2021-04-09 12-15-08

**Comment by Temirtator** [comment]
@lekoaf  ah yes, sorry for disturbing

**Comment by amcasey** [comment]
@Temirtator if this is primarily for your own consumption, I would agree with @lekoaf's suggestion that you just add index.d.ts directly to your fork.  And you don't need to ping people when you create a PR here - the bot will take care of that.  (Exception: if you happen to know that someone is specifically interested, e.g. because of a conversation on another GH issue.)

**Pull request closed without merging**